Fix double event fired by QSpinBox in slow graduated

elpaso opened the pull request:
... renderer widget

Fixes #31635

For the record of the underlying issue:

https://lists.qt-project.org/pipermail/interest/2013-July/007936.html
https://forum.qt.io/topic/82181/qt-doublespin-box-value-changed-slot-has-called-twice
https://forum.qt.io/topic/96094/qspinbox-value-changed-slot-has-called-twice-on-mouse-click
https://www.qtcentre.org/threads/53709-slot-valueChanged(double)-of-doubleSpinBox-called-twice-if-breakpoint-is-set

elpaso committed:
Fix double event fired by QSpinBox in slow graduated

... renderer widget

Fixes #31635

For the record of the underlying issue:

https://lists.qt-project.org/pipermail/interest/2013-July/007936.html
https://forum.qt.io/topic/82181/qt-doublespin-box-value-changed-slot-has-called-twice
https://forum.qt.io/topic/96094/qspinbox-value-changed-slot-has-called-twice-on-mouse-click
https://www.qtcentre.org/threads/53709-slot-valueChanged(double)-of-doubleSpinBox-called-twice-if-breakpoint-is-set

nyalldawson commented on src/gui/symbology/qgsgraduatedsymbolrendererwidget.cpp:
```diff
@@ -906,6 +910,8 @@ void QgsGraduatedSymbolRendererWidget::symmetryPointEditingFinished( )
 void QgsGraduatedSymbolRendererWidget::classifyGraduated()
 {
+
+  QApplication::setOverrideCursor( Qt::WaitCursor );
```
There's a scoped class in GUI which handles this nicely

nyalldawson replied:
(QgsTemporaryCursorOverride)

elpaso replied:
Thanks! I was pretty sure that there was a scoped helper but I didn't search hard enough.
Do you think we could collect all this little helpers in a dedicated doxy section?

nyalldawson commented:
Gosh, that's pretty... yuck. Oh well. 

I'm guessing QgsDoubleSpinBox should have the same workaround?

elpaso commented:
> Gosh, that's pretty... yuck. Oh well.
> 
> I'm guessing QgsDoubleSpinBox should have the same workaround?

I think so, it also inherits from QAbstractSpinBox. Would a mixin base class work for us in this case?

nyalldawson commented:
> I think so, it also inherits from QAbstractSpinBox. Would a mixin base class work for us in this case?

Not in c++ it won't :P

elpaso commented:
I'm still eating too much Python these days ;)

elpaso committed:
Use scoped class for wait cursor

elpaso committed:
Apply timer spinbox workaround to QgsDoubleSpinBox too

To prevent double changed event fired in case the slot
takes too long to execute.
